src/main/java/de/tum/in/www1/artemis/service/competency/CompetencyProgressService.java

```diff
@@ -154,20 +151,26 @@ else if (learningObject instanceof LectureUnit lectureUnit) {
     }
 
     /**
-     * Updates the progress value (and confidence score) of the given competency and user, then returns it
+     * Updates the progress value and confidence score of the given competency and user, then returns it
      *
      * @param competencyId The id of the competency to update the progress for
      * @param user         The user for which the progress should be updated
      * @return The updated competency progress, which is also persisted to the database
      */
     public CompetencyProgress updateCompetencyProgress(Long competencyId, User user) {
-        var competency = competencyRepository.findByIdWithExercisesAndLectureUnitsAndCompletions(competencyId).orElse(null);
+        Optional<Competency> optionalCompetency = competencyRepository.findByIdWithLectureUnits(competencyId);
 
-        if (user == null || competency == null) {
+        if (user == null || optionalCompetency.isEmpty()) {
             log.debug("User or competency no longer exist, skipping.");
             return null;
         }
 
+        Competency competency = optionalCompetency.get();
+        Set<LectureUnit> lectureUnits = competency.getLectureUnits().stream().filter(lectureUnit -> !(lectureUnit instanceof ExerciseUnit)).collect(Collectors.toSet());
```

Comment:
Wouldn't that be a more robust implementation?

```suggestion
        Set<LectureUnit> lectureUnits = competency.getLectureUnits().stream().filter(unit -> unit instanceof LectureUnit).collect(Collectors.toSet());
```

If I am not mistaken this would otherwise break if another unit type is added

Reply:
`LectureUnit` is the super class, so `ExerciseUnit` also fulfils your suggested change. We want every subclass except ExerciseUnits, since they are already covered by the actual exercises.
Other new subclasses should also be considered, so there shouldn't be a problem in the future